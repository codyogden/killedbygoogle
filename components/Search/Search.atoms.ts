import styled from '@emotion/styled';
import { css } from '@emotion/react';

export const SearchContainer = styled.div(() => css({
  boxSizing: 'border-box',
}));

export const SearchBox = styled.input(() => css({
  backgroundColor: 'transparent',
  border: 'none',
  borderBottom: '1px solid #cfcfcf',
  boxSizing: 'border-box',
  display: 'block',
  fontSize: '1.25em',
  fontWeight: 'lighter',
  height: '100%',
  margin: '0 auto',
  outline: 'none',
  transition: 'all 100ms linear',
  width: '100%',

  ['@media screen and (max-width: 700px)']: {
    fontSize: '1em',
  },

  ['::-webkit-input-placeholder']: {
    color: '#cfcfcf',
    transition: 'all 100ms linear',
  },

  ['::-moz-placeholder']: {
    color: '#cfcfcf',
    transition: 'all 100ms linear',
  },

  [':-ms-input-placeholder']: {
    color: '#cfcfcf',
    transition: 'all 100ms linear',
  },

  [':-moz-placeholder']: {
    color: '#cfcfcf',
    transition: 'all 100ms linear',
  },

  ['&:hover']: {
    ['::-webkit-input-placeholder']: {
      color: '#cfcfcf',
    },

    ['::-moz-placeholder']: {
      color: '#cfcfcf',
    },

    [':-ms-input-placeholder']: {
      color: '#cfcfcf',
    },

  },
  ['&:focus']: {

    ['::-webkit-input-placeholder']: {
      color: '#cfcfcf',
    },

    ['::-moz-placeholder']: {
      color: '#cfcfcf',
    },

    [':-ms-input-placeholder']: {
      color: '#cfcfcf',
    },

    [':-moz-placeholder']: {
      color: '#cfcfcf',
    },
  },
}));
