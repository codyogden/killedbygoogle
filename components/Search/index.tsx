/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useEffect, useState } from 'react';
import SRT from 'components/SRT';

// Import Styled Components
import { SearchContainer, SearchBox } from 'components/Search/Search.atoms';

interface SearchProps {
  searchCallback: Function
}

export default function Search({ searchCallback }: SearchProps) {
  const [searchValue, updateValue] = useState('');

  useEffect(() => {
    const delayFn = setTimeout(() => searchCallback(searchValue), 350);
    return () => clearTimeout(delayFn);
  }, [searchValue, searchCallback]);

  return (
    <SearchContainer>
      <label htmlFor="searchBox" aria-label="Search">
        <SRT>Search</SRT>
        <SearchBox
          id="searchBox"
          placeholder="Search"
          type="text"
          value={searchValue}
          onChange={(event: React.FormEvent<HTMLInputElement>) => updateValue(event.currentTarget.value)}
        />
      </label>
    </SearchContainer>
    );
}
