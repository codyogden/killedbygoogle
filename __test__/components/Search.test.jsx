import React from 'react';
import { shallow } from 'enzyme';
import { SearchBox } from '../../src/components/Search.atoms';
import Search from '../../src/components/Search';

describe('Search Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Search search={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('typing in the SearchBox properly updates state and invokes callback function', () => {
    const searchSpy = jest.fn();
    const wrapper = shallow(
      <Search search={searchSpy} />,
    );
    const event = { target: { value: 'hello' } };

    wrapper.find(SearchBox).simulate('change', event);
    expect(wrapper.state().inputValue).toEqual('hello');
    expect(searchSpy).toHaveBeenCalledWith('hello');
  });
});
