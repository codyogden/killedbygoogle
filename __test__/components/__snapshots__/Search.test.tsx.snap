// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches snapshot 1`] = `
<styled.div>
  <label
    aria-label="Search"
    htmlFor="searchBox"
  >
    <span
      style={
        Object {
          "display": "none",
        }
      }
    >
      Search
    </span>
    <styled.input
      id="searchBox"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </label>
</styled.div>
`;
